web/impact/impact/v1/views/program_family_detail_view.py

```diff
@@ -0,0 +1,33 @@
+# MIT License
+# Copyright (c) 2017 MassChallenge, Inc.
+
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework_tracking.mixins import LoggingMixin
+
+from impact.permissions import (
+    V1APIPermissions,
+)
+from impact.models import ProgramFamily
+from impact.v1.helpers import ProgramFamilyHelper
+from impact.v1.metadata import ImpactMetadata
+
+
+class ProgramFamilyDetailView(LoggingMixin, APIView):
+    model = ProgramFamily
+    metadata_class = ImpactMetadata
+
+    permission_classes = (
+        V1APIPermissions,
+    )
+
+    METADATA_ACTIONS = {
+        "GET": ProgramFamilyHelper.DETAIL_METADATA
```

Comment:
I fee like these metadata constants are a bit all over the place - wont make this a blocker but im seeing that some of the metadatas are coming off helper classes, some only have a DETAIL_METADATA that is used for all views (post/get/list) - it might be worth wrangling this in for consistency.

Reply:
The METADATA_ACTIONS information needs to be findable from the view (see v1.metadata.ImpactMetadata) and it varies based on the view (UserListView has different actions than UserHistoryView).  However, there is information that is shared between views (ProgramFamilyHelper.DETAIL_METADATA is used by both UserListView and UserDetailView).  I don't want to import this data from one view to another, so putting the object detail descriptions in the helper class seems logical to me.  Ideally more of this data would be derived from the object, but that gets into the black list/white list issue and the metadata needed to really implement OPTIONS correctly.  I don't see a way to more cleanly handle this that doesn't involve significant development work that would be done better once we have a standard.